src/uct/api/uct.h

```diff
@@ -340,11 +340,29 @@ enum uct_md_mem_flags {
                                                 mapping may be deferred until
                                                 it is accessed by the CPU or a
                                                 transport. */
-    UCT_MD_MEM_FLAG_FIXED    = UCS_BIT(1)  /**< Place the mapping at exactly
+    UCT_MD_MEM_FLAG_FIXED    = UCS_BIT(1), /**< Place the mapping at exactly
                                                 defined address */
+    /* memory access flags */
+    UCT_MD_MEM_ACCESS_LOCAL         = UCS_BIT(2), /**< enable local read/write
+                                                       access */ 
+    UCT_MD_MEM_ACCESS_REMOTE_WRITE  = UCS_BIT(3), /**< enable remote read 
+                                                       access */
+    UCT_MD_MEM_ACCESS_REMOTE_READ   = UCS_BIT(4), /**< enable remote write 
+                                                       access */
+    UCT_MD_MEM_ACCESS_REMOTE_ATOMIC = UCS_BIT(5)  /**< enable remote atomic 
+                                                       access */
 };
 
+#define UCT_MD_MEM_ACCESS_ALL       (UCT_MD_MEM_ACCESS_LOCAL|\
+                                     UCT_MD_MEM_ACCESS_REMOTE_WRITE|\
+                                     UCT_MD_MEM_ACCESS_REMOTE_READ|\
+                                     UCT_MD_MEM_ACCESS_REMOTE_ATOMIC)
 
+#define UCT_MD_MEM_ACCESS_RMA       (UCT_MD_MEM_ACCESS_LOCAL|\
+                                     UCT_MD_MEM_ACCESS_REMOTE_WRITE|\
+                                     UCT_MD_MEM_ACCESS_REMOTE_READ)
+
+#define UCT_MD_MEM_ACCESS_DEFAULT    UCT_MD_MEM_ACCESS_ALL
```

Comment:
why would we need a concept of "default"? user has to pass which access mode is required
for now we added ALL to tests to preserve previous behavior